src/mtype.h

```diff
@@ -268,8 +271,15 @@ class Type : public RootObject
     virtual bool isscope();
     virtual bool isString();
     virtual bool isAssignable();
+
+    virtual TypeStruct* isTypeStruct() { return NULL; }
+    virtual TypeClass* isTypeClass() { return NULL; }
+    virtual bool isTypeAggregate() { return this->isTypeStruct() || this->isTypeClass(); }
+    virtual TypeDArray* isTypeDArray() { return NULL; }
```

Comment:
But usage of `isTypeStruct()´ in if-expression is easier to read. If you're afraid of virtual functions being too slow why not defined and use (inline) wrappers around type->ty == Txxxx plus cast.

Please consider this. I'm trying to make the DMD code base easier to understand and work with, especially for newcomers (like me).